### ISSUE TITLE
Question on regex filtering

### ISSUE THREAD

**Issue by jrosebr1** [realbig/PL_PYIS_HELPSCOUT_DRIP#6]
Just a quick question here RegEx filtering:

What PHP (or JavaScript?) function call is being used for RegEx?

From the attached screenshots you can see that a I have a "customer" tag which also pulls down the "gurus non-customers segment 2" tag.

![screen shot 2017-03-01 at 5 01 08 pm](https://cloud.githubusercontent.com/assets/759645/23508924/c8b0dda4-ff21-11e6-9447-75b10a010827.png)
![screen shot 2017-03-01 at 5 01 18 pm](https://cloud.githubusercontent.com/assets/759645/23508926/c8b1a342-ff21-11e6-8d08-9bf1541e0906.png)

Basically, I need to create my RegEx filter: `/$customer^/`, but depending on which function is performing the filter (and how the filters are escaped and stored in the database), I need to know how to format the filters.

**Comment by d4mation** [comment]
Ah, I'm just doing a simple [`preg_match()`](http://php.net/manual/en/function.preg-match.php), but I'm including the two slashes on either side within the code (And the matches are all case-insensitive since I threw the `i` flag into the code itself).

So in this case, you would just need to enter `$customer^`.

**Comment by d4mation** [comment]
I did not think about instances where "non-regex" string would get handled like this though.

Would it be better if I removed the slashes and had you enter those manually so that I could more accurately handle this? That way `customer` wouldn't match `gurus non-customers segment 2`, but `/.+customer.+/` (bad regex) would?

**Comment by jrosebr1** [comment]
I think since we're already going down the rabbit hole with regex, let's make the "tag" feature 100% regex. That way I can always customize the regex string and not have to ping you guys for a feature update. Does that work for you?

**Comment by d4mation** [comment]
Right now it is 100% regex, it just just set up like this:

`/<tag_goes_here>/i`

That's why `gurus non-customers segment 2` got picked up by `customer` as it was actually processing `/customer/i`.

**Comment by jrosebr1** [comment]
Thanks for the clarification! I've added a bunch of new regex filters to the system and will be testing them out this weekend. I'll get back to you early next week on any regex filtering. I have any suggestion, but I'll open a separate thread for it.

**Comment by jrosebr1** [comment]
The regex filtering is working as expected, closing this.
